### PR TITLE
Separating game setup from game class

### PR DESCRIPTION
**Pull request by RabeaGleissner**
@felipesere @jsuchy

I extracted the two methods that set up the game from the `Game` class into a `GameSetup` class. 

I found it hard to test if `GameSetup` creates the correct players but ended up storing the new game object in a field variable so that the `GameSetup` tests can access the new game and check if it has the correct player and opponent. 

Do you think this is a good way of doing it? I'm a bit concerned that other classes in the package can access the `getPlayer()` and `getOpponent()` methods. Also, I now have two getter methods in `GameSetup` and two more in `Game` plus the field variable in `GameSetup` - all purely for testing purposes. Is this too much production code that only serves tests?

Thank you!

### CODE REVIEW

File: src/main/java/de/rabea/game/Game.java [hunk 1]
```diff
@@ -46,7 +27,7 @@ public void play(int boardDimension) {
         }
         finishGame(currentPlayer.mark(), board);
         if (userInterface.playAgain()) {
-            setUpNewGame();
+            gameSetup.setUpNewGame();
```
Comment (felipesere) [inline src/main/java/de/rabea/game/Game.java:30]: Why would the game need to start 'a new game'?
Is the `Game` class responsible for running _a single game_ or _all of them_?

File: src/test/java/de/rabea/game/GameSetupTest.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        GameSetup gameSetup = new GameSetup(fakeUserInterface, playerFactory);
+        gameSetup.startApplication();
+        assertTrue(fakeUserInterface.greetUserWasCalled);
+        assertTrue(gameSetup.getPlayer() instanceof HumanPlayer);
+        assertTrue(gameSetup.getOpponent() instanceof HumanPlayer);
+    }
+
+    @Test
+    public void setsUpFirstHumanVsComputer3x3Game() {
+        fakeUserInterface.provideConsoleInput("2", "3", "1", "2", "3", "n");
```
Comment (felipesere) [inline src/test/java/de/rabea/game/GameSetupTest.java:36]: This test is pretty long and hard to read. Is there a simpler way to do the setup?

File: src/main/java/de/rabea/game/Game.java (hunk 1 above)
Comment (RabeaGleissner) [inline src/main/java/de/rabea/game/Game.java:30]: I don't think I understand what you mean here... In this case the `Game` is asking the `GameSetup` class to set up the new game. Or do you mean that `Game` shouldn't know if the user wants to play again or not?

Comment (RabeaGleissner) [inline src/main/java/de/rabea/game/Game.java:30]: @felipesere OK, ignore my previous comment... I spoke to Jim about this bit of the code. I have a story to get rid of the circular dependency between Game and GameSetup, so I think that will eliminate this code smell.

**Pull request merged**